docs/src/main/tut/datatypes/approx/bloom_filter.md

```diff
@@ -8,16 +8,50 @@ scaladoc: "#com.twitter.algebird.BloomFilter"
 
 # Bloom Filter
 
+A Bloom filter is a data structure invented by Burton Howard Bloom in 1970. It makes it possible to check for set
+membership in a space efficient way. It does in a probabilistic manner, probiding the property that there may be false
+ positives, but its guaranteed to have no false negatives.
+
+A Bloom filter is created by initializing a bit array of n bits, with all bits set to 0. Elements are then added by
+computing m hash functions which output values correspond to the indexes in the byte array with a uniform random
+distribution. As these hash functions are computed, the bytes in the array corresponding to the hash function output
+are set to 1. This is done for all elements added to the set.
+
+To query the filter for membership of a element, the same m hash functions are computed for that element, and the
+corresponding bits are checked. If all bits are not set to 1, we know that the element was not part of the set. However,
+if all bits are set to one the element might be a member of the set.
+
+The size of a Bloom filter depends on the number of input elements, and the desired false positive rate.
+
+To read more about Bloom filters see wikipedia: https://en.wikipedia.org/wiki/Bloom_filter
+
 example usage:
 
 ```tut:book
 import com.twitter.algebird._
+
+// We need to provide a implicit conversion from of the Bloom filter,
```

Comment:
Aha. I see. I had missed specifying the type for `BloomFilterMonoid` here which meant that I god a compilation error about multiple type classes matching. But specifying the type solved that.